turtle_oxford.py

```
@@ -871,11 +918,750 @@ def qint(s: str, mult: int, default: int) -> int:
+
+def keybuffer(size: int):
+    """Create a key buffer of the specified size.
+
+    :param size: the size of the key buffer
+    :type size: int
+    """
+    TurtleCanvas._key_buffer_size = size
+    TurtleCanvas._key_buffer = []
+
+def keyecho(on: bool):
+    """Turn on or off key echo to console
+
+    :param on: True to turn on key echo, False to turn it off
+    :type on: bool
+    """
+    TurtleCanvas._key_echo = on
+
+def read(max_size: int) -> str:
+    """Read a string from the keyboard buffer of the specified size.
+
+    :param max_size: the maximum size of the string to read
+    :type max_size: int
+    :return: the string read from the keyboard buffer
+    :rtype: str
+    """
+    if len(TurtleCanvas._key_buffer) == 0:
+        return ""
+    return TurtleCanvas._key_buffer[:min(max_size, len(TurtleCanvas._key_buffer))] # Return the first max_size characters
+
+def mkdir(name: str) -> bool:
+    """Create a new directory with the specified name.
+
+    :param name: the name of the new directory
+    :type name: str
+    :return: True if the directory was created successfully, False otherwise
+    :rtype: bool
+    """
+    try:
+        os.mkdir(name)
+        return True
+    except Exception:
+        return False
+    
+def rmdir(name: str) -> bool:
+    """Remove the directory with the specified name.
+
+    :param name: the name of the directory to remove
+    :type name: str
+    :return: True if the directory was removed successfully, False otherwise
+    :rtype: bool
+    """
+    try:
+        os.rmdir(name)
+        return True
+    except Exception:
+        return False
+
+def mkfile(name: str) -> bool:
+    """Create a new file with the specified name.
+
+    :param name: the name of the new file
+    :type name: str
+    :return: True if the file was created successfully, False otherwise
+    :rtype: bool
+    """
+    try:
+        open(name, "w").close()
+        return True
+    except Exception:
```

open will return `True` for existing files, too, if the user running the process has write permissions to it. Also, this effectively silences the error, which is bad practice because it's impossible to debug. Try redirecting the exception message to stdout or stderr.